docs/rules/jsx-curly-newline.md

```diff
@@ -0,0 +1,149 @@
+# Enforce linebreaks in curly braces in JSX attributes and expressions. (react/jsx-curly-newline)
+
+Many style guides require or disallow newlines inside of jsx curly expressions.
+
+**Fixable:** This rule is automatically fixable using the `--fix` flag on the command line.
+
+## Rule Details
+
+This rule enforces consistent linebreaks inside of curlies of jsx curly expressions.
+
+## Rule Options
+
+This rule accepts either an object option:
+
+```ts
+{
+  multiline: "consistent" | "forbid" | "require", // default to 'consistent'
+  singleline: "consistent" | "forbid" | "require", // default to 'consistent'
+}
+```
+Option `multiline` takes effect when the jsx expression inside the curlies occupies multiple lines.
+
+Option `singleline` takes effect when the jsx expression inside the curlies occupies a single line.
+
+* `consistent` enforces either both of linebreaks around curlies are present, or none are present.
```

Comment:
i think "around curlies" might not be the proper term, since that implies "surrounding the curlies" (outside) as opposed to "inside". Maybe:
```suggestion
* `consistent` enforces either both curly braces have a line break directly inside them, or no line breaks are present.
```